src/utilities/__tests__/stripIgnoredTokens-test.js

```diff
@@ -0,0 +1,461 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { parse } from '../../language/parser';
+import { print } from '../../language/printer';
+import dedent from '../../jsutils/dedent';
+import { stripIgnoredTokens } from '../stripIgnoredTokens';
+import {
+  kitchenSinkQuery,
+  kitchenSinkSDL,
+  strippedKitchenSinkQuery,
+  strippedKitchenSinkSDL,
+} from '../../__fixtures__';
+
+describe('stripIgnoredTokens: Query document', () => {
+  const kitchenSink = kitchenSinkQuery;
+
+  it('does not alter ast', () => {
+    const ast = parse(kitchenSink);
+    const astBefore = JSON.stringify(ast);
+    stripIgnoredTokens(print(ast));
+    expect(JSON.stringify(ast)).to.equal(astBefore);
+  });
+
+  it('prints minimal ast', () => {
+    const ast = { kind: 'Field', name: { kind: 'Name', value: 'foo' } };
+    expect(stripIgnoredTokens(print(ast))).to.equal('foo');
+  });
+
+  it('produces helpful error messages', () => {
+    const badAst1 = { random: 'Data' };
+    expect(() => stripIgnoredTokens(print(badAst1))).to.throw(
+      'Invalid AST Node: { random: "Data" }',
+    );
+  });
+
+  it('correctly prints non-query operations without name', () => {
+    const queryAstShorthanded = parse('query { id, name }');
```

Comment:
@rybon You basically duplicated all tests from `print`.
Sadly most of them not relevant to this functionality and you basically test `print` one more time and not covering edge cases specific to `stripIgnoredTokens`.
These test shouldn't contain `print` at all since `stripIgnoredTokens` work with string, not with AST.

I will work on this tomorrow.